zio-http/shared/src/main/scala/zio/http/codec/StringCodec.scala

```diff
@@ -0,0 +1,394 @@
+package zio.http.codec
+
+import java.time._
+import java.util.{Currency, UUID}
+
+import scala.annotation.tailrec
+
+import zio._
+
+import zio.stream._
+
+import zio.schema._
+import zio.schema.annotation.simpleEnum
+import zio.schema.codec._
+
+import zio.http.Charsets
+
+object StringCodec {
+  type StringCodec[A] = Codec[String, Char, A]
+  private def errorCodec[A](schema: Schema[A]) =
+    new Codec[String, Char, A] {
+      override def decode(whole: String): Either[DecodeError, A] = throw new IllegalArgumentException(
+        s"Schema $schema is not supported by StringCodec.",
+      )
+
+      override def streamDecoder: ZPipeline[Any, DecodeError, Char, A] = throw new IllegalArgumentException(
+        s"Schema $schema is not supported by StringCodec.",
+      )
+
+      override def encode(value: A): String = throw new IllegalArgumentException(
+        s"Schema $schema is not supported by StringCodec.",
+      )
+
+      override def streamEncoder: ZPipeline[Any, Nothing, A, Char] = throw new IllegalArgumentException(
+        s"Schema $schema is not supported by StringCodec.",
+      )
+    }
+
+  @tailrec
+  private def emptyStringIsValue(schema: Schema[_]): Boolean = {
+    schema match {
+      case value: Schema.Optional[_] =>
+        val innerSchema = value.schema
+        emptyStringIsValue(innerSchema)
+      case _                         =>
+        schema.asInstanceOf[Schema.Primitive[_]].standardType match {
+          case StandardType.UnitType   => true
+          case StandardType.StringType => true
+          case StandardType.BinaryType => true
+          case StandardType.CharType   => true
+          case _                       => false
+        }
+    }
+  }
+
+  implicit def fromSchema[A](implicit schema: Schema[A]): Codec[String, Char, A] = {
```

Comment:
Can we do this whole thing as `Schema#toString(value)` and `Schema#fromString(string)`?? Why does this even belong here?